tests/plugins/test_amp_plugins.py

```diff
@@ -0,0 +1,81 @@
+import os
+from unittest import mock
+
+import pytest
+import torch
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.plugins import ApexMixedPrecisionPlugin, NativeMixedPrecisionPlugin
+from tests.helpers import BoringModel
+from tests.helpers.runif import RunIf
+
+
+class MyNativeAMP(NativeMixedPrecisionPlugin):
+    pass
+
+
+class MyApexPlugin(ApexMixedPrecisionPlugin):
+    pass
+
+
+@mock.patch.dict(
+    os.environ, {
+        "CUDA_VISIBLE_DEVICES": "0,1",
+        "SLURM_NTASKS": "2",
+        "SLURM_JOB_NAME": "SOME_NAME",
+        "SLURM_NODEID": "0",
+        "LOCAL_RANK": "0",
+        "SLURM_LOCALID": "0",
+    }
+)
+@mock.patch('torch.cuda.device_count', return_value=2)
+@pytest.mark.parametrize('ddp_backend,gpus', [('ddp', 2), ('ddp2', 2), ('ddp_spawn', 2)])
+@pytest.mark.parametrize(
+    'amp,plugin_arg,plugin_cls', [
+        pytest.param('native', None, NativeMixedPrecisionPlugin, marks=RunIf(amp_native=True)),
+        pytest.param('native', [MyNativeAMP()], MyNativeAMP, marks=RunIf(amp_native=True)),
+        pytest.param('apex', None, ApexMixedPrecisionPlugin, marks=RunIf(amp_apex=True)),
+        pytest.param('apex', [MyApexPlugin(amp_level="O2")], MyApexPlugin, marks=RunIf(amp_apex=True))
+    ]
+)
+def test_amp_apex_ddp(mocked_device_count, ddp_backend, gpus, amp, plugin_arg, plugin_cls):
+
+    trainer = Trainer(
+        fast_dev_run=True,
+        precision=16,
+        amp_backend=amp,
+        gpus=gpus,
+        accelerator=ddp_backend,
+        plugins=plugin_arg,
+    )
+    assert isinstance(trainer.precision_plugin, plugin_cls)
+
+
+class GradientUnscaleBoringModel(BoringModel):
+
+    def on_after_backward(self):
+        norm = torch.nn.utils.clip_grad_norm_(self.parameters(), 2)
+        if not (torch.isinf(norm) or torch.isnan(norm)):
+            assert norm.item() < 15.
+
+
+@RunIf(min_gpus=2, amp_native=True)
+@pytest.mark.parametrize('accum', [1, 2])
```

Comment:
the only real diff before was nb accumulated batches, the classes were identical